OfflineMapAreas Github docs (#943)
Esri/arcgis-maps-sdk-kotlin-toolkit: toolkit/offline/README.md

|:---:	|:---:	|
| <img src="preplanned_screenshot.png" height=500> 	| <img src="ondemand_screenshot.png" height=500> 	|

## Features

Supports both ahead-of-time (preplanned) and on-demand map areas for an offline enabled web map. This `OfflineMapAreas` composable:

- Displays a list of map areas.
- Shows download progress and status for map areas.
- Opens a map area for viewing when selected.
- Provides options to view details about downloaded map areas.
- Supports removing downloaded offline map areas files from the device.

For preplanned workflows, this composable:
- Displays a list of available preplanned map areas from an offline-enabled web map that contains preplanned map areas when the network is connected.
- Downloads preplanned map areas in the list.
- Displays a list of downloaded preplanned map areas on the device when the network is disconnected.

For on-demand workflows, this composable:
- Allows users to add and download on-demand map areas to the device by specifying an area of interest and level of detail.
- Displays a list of on-demand map areas available on the device that are tied to a specific web map.

> ## Usage

Reviewer: We should add information about the permissions that need to be added to the APP manifest to make the component work here.

To use the composable you would need to provide an `OfflineMapState`. The state object provides two constructors to initialize the `OfflineMapAreas` composable using an `ArcGISMap` or an `OfflineMapInfo`. Therefore, the composable can be used either when the device is connected to or disconnected from the network.

In other words, the composable can be used in the following situations:
- When the device is connected to the network…
  - Displays preplanned map areas from a web map that are available for download.
  - When the web map doesn’t contain preplanned map areas, users can add and download on-demand map areas by specifying a geographic area and level of detail.
  - Use the constructor with `ArcGISMap` to create the state.
- When the device is disconnected from the network…
  - Displays only downloaded map areas by retrieving offline map info from the device.

>   - Use the constructor with `OfflineMapInfo` to create the state.

Reviewer: We can add information here about how to get the offlineMapInfo to initialize the state with it(via the repository)

- Downloads preplanned map areas in the list.
- Displays a list of downloaded preplanned map areas on the device when the network is disconnected.

For on-demand workflows, this composable:
- Allows users to add and download on-demand map areas to the device by specifying an area of interest and level of detail.
- Displays a list of on-demand map areas available on the device that are tied to a specific web map.

## Usage

To use the composable you would need to provide an `OfflineMapState`. The state object provides two constructors to initialize the `OfflineMapAreas` composable using an `ArcGISMap` or an `OfflineMapInfo`. Therefore, the composable can be used either when the device is connected to or disconnected from the network.

In other words, the composable can be used in the following situations:
- When the device is connected to the network…
  - Displays preplanned map areas from a web map that are available for download.
  - When the web map doesn’t contain preplanned map areas, users can add and download on-demand map areas by specifying a geographic area and level of detail.
  - Use the constructor with `ArcGISMap` to create the state.

> - When the device is disconnected from the network…

Reviewer: Offline WorkFlow (Device is disconnected from the network)

  - Displays only downloaded map areas by retrieving offline map info from the device.

>   - Use the constructor with `OfflineMapInfo` to create the state. The list of all portal item information for web maps that have downloaded map areas can be obtained using the `OfflineRepository` singleton object. 

Reviewer: ```suggestion
  - Use the constructor with `OfflineMapInfo` to create the state. `OfflineRepository.OfflineMapInfos` provides the list of all portal item information for web maps that have downloaded map areas can be obtained using the `OfflineRepository` singleton object. 
```

- When the device network connection has changed…
  - Re-initialize the the state `OfflineMapState` using the desired constructor.

> 

Reviewer: We should add Permissions heading here

### Permissions

see https://github.com/Esri/arcgis-maps-sdk-kotlin-toolkit/tree/main/toolkit/featureforms#permissions

> The `OfflineMapAreas` toolkit component requires the following app-level permissions:

Reviewer: ```suggestion
The `OfflineMapAreas` toolkit component requires the following app-level permissions to function correctly. These permissions are added to the AndroidManifest.xml file of the app.
```

- `android.permission.FOREGROUND_SERVICE_DATA_SYNC`: This permission is needed to enable the  component to perform data synchronization operations, such as downloading large map areas, as a foreground service, ensuring these downloads continue even if app is backgrounded or closed.

- `android.permission.INTERNET`: This permission is needed as the component would connect to the network to display available preplanned map areas from a web map and to download both preplanned and on-demand map areas.

> - `android.permission.POST_NOTIFICATIONS`: This permission is used to allow the component to display notifications, to show the progress and status of map area downloads, or to notify when a download is complete.

Reviewer: ```suggestion
- `android.permission.POST_NOTIFICATIONS`: This permission is essential for enabling the component to display notifications. It allows the component to communicate the progress and status of map area downloads, as well as to notify users upon the completion of a download.
```
